src/sentry/migrations/0110_sentry_app_creator_backill.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-10-06 17:57
+from __future__ import unicode_literals
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+from django.db import migrations
+
+def backfill_one(audit_log_entry, SentryApp):
+    name = audit_log_entry.data.get("sentry_app")
+    if name:
+        # find the sentry app for that org, name, that's not deleted
+        sentry_app = SentryApp.objects.filter(
+            name=name,
+            owner=audit_log_entry.organization,
+            date_deleted__isnull=True,
+            creator_user__isnull=True,
+        ).first()
+        # if there is a match and the user exists,
+        # update with the creator
+        user = audit_log_entry.actor
+        if sentry_app and user:
+            sentry_app.creator_user = user
+            sentry_app.creator_label = user.email or user.username
+            sentry_app.save()
+
+
+def backfill_sentry_app_creator(apps, schema_editor):
+    """
+    Backills the creator fields of SentryApp from
+    the audit log table
+    """
+    SentryApp = apps.get_model("sentry", "SentryApp")
+    AuditLogEntry = apps.get_model("sentry", "AuditLogEntry")
+
+    queryset = AuditLogEntry.objects.filter(event=113) # sentry app add
+    for audit_log_entry in RangeQuerySetWrapperWithProgressBar(queryset):
+        backfill_one(audit_log_entry, SentryApp)
```

Comment:
Since `AuditLogEntry` is pretty large (20mm rows) and `event` isn't indexed, I'd suggest flipping this around - Iterate through `SentryApp` rows where `creator_user__isnull=True, date_deleted__isnull=True` and then search for related `AuditLogEntry` rows from the related org. That'll be quicker since you can just fetch all `AuditLogEntry` rows for that org, filtered to the `event` you want.

Reply:
Lol, I posted this before I saw the other comments, looks like we were heading in this direction already.